config/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,115 @@
+package v1alpha2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsDataGather provides data gather configuration options for the the Insights Operator.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=insightsdatagathers,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1245
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=config-operator,operatorOrdering=01
+// +openshift:enable:FeatureGate=InsightsConfig
+// +openshift:compatibility-gen:level=4
+type InsightsDataGather struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// spec holds user settable values for configuration
+	// +required
+	Spec InsightsDataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsDataGatherStatus `json:"status"`
+}
+
+type InsightsDataGatherSpec struct {
```

Comment:
This spec has no required fields, what does it mean when the spec is empty?

Reply:
AFAIK the InsightsDataGather CRD is an optional configuration of the Insigths Operator. If there are no values in the spec we will use defaults. Please correct me if I am wrong here @tremes

Reply:
Will this object always exist, to show status? Or does something happen only when the resource exists, and we expect the customer to create the resource?

Reply:
This was designed as a `cluster` singleton CR if I remember correctly. So it's created by the operator (CVO) and is used for the Insights configuration (by user). My understanding is that the status is reported in `insightsoperator.operators.openshift.io`

Reply:
I think it was the general approach back then. I mean things that can be configured live in the `config.openshift.io` group and things that are reported by the operators live in the `operators.openshift.io`. That was the general consensus, or at least I understood it. 
I also remember that the `insightsoperator.operators.openshift.io` (the CR to report status) was immediately introduced as `v1` and I guess we weren't so sure about the config options so the `insightsdatagather` started as `v1alpha1`.